CMakeLists.txt

```diff
@@ -71,13 +51,30 @@ if(DEFINED ENV{ROS_VERSION} )
       "srv/RunCommand.srv"
       "srv/RunPythonCommand.srv"
       "srv/RunPythonFile.srv"
-      )
-
-    #    install(TARGETS ${PROJECT_NAME} EXPORT ${TARGETS_EXPORT_NAME} DESTINATION lib)
+    )
     install(FILES package.xml DESTINATION share/${PROJECT_NAME})
-
     ament_export_dependencies(rosidl_default_runtime)
     ament_package()
+  endif()
 
+  if(BUILD_TESTING)
+    include(CTest)
+    find_program(py_executable "python3")
```

Comment:
I think this should be REQUIRED. 
And in that case, we can remove the following if

Reply:
https://cmake.org/cmake/help/latest/command/find_program.html
REQUIRED argument is only available from CMake 3.18 version.
- Ubuntu 18.04 CMake default version is still 3.5 (if I recall properly)
- Ubuntu 20.04 CMake version is 3.16.3

Hence the "if".

Reply:
Ok, but my issue is mostly that if BUILD_TESTING is activated and no python interpreter is found, we shoudn't silently pass without running anything,

Maybe you can use:
```cmake
include(cmake/python.cmake)
FINDPYTHON()
```

And this will provide `$PYTHON_EXECUTABLE`, or raise the proper `FATAL_ERROR` if anything went wrong.